src/main/java/org/tikv/common/region/AbstractRegionStoreClient.java

```diff
@@ -155,6 +156,42 @@ public boolean onStoreUnreachable(BackOffer backOffer) {
     return false;
   }
 
+  public ByteString buildRequestKey(ByteString key) {
+    switch (conf.getApiVersion()) {
```

Comment:
can we get the cluster version and tikv configuration from PD? so that we don't need to add one more config in the java client and the java client can automatically adjust the encode/decode methods according to the cluster it is connected to.

Reply:
By design, the client should not read the TiKV configuration from PD or themself. Because:

1. TiKV configurations could be different from each other due to the restart process.
2. TiDB use client-go, which should have the same behaviour as the client-java, however, TiDB would never use API v2 but use API v1 instead, due to they have a key prefix with `m` or `t`. Thus, client-go should not read the API version from the cluster components, instead, just set the version to `1`.